### PR TITLE
Update user story and use case

### LINKED ISSUE #13: Update developer guide user story and use case

### PR DESCRIPTION
**Pull request by wallacexuhanxiao**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -307,13 +342,81 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 3a1. PedagoguePages shows an error message.
+
+      Use case resumes at step 2.
+    
+* 3b. The updated field is in a wrong format
+    * 3b2. PedagoguePages shows an error message.
+
+      Use case resumes at step 2.
+
+**Use case: find a group of a students by tag**
```
Comment (IamZhenHong) [inline docs/DeveloperGuide.md:396]: Perhaps adding a new extension for when the tag input is invalid?

**Review by IamZhenHong — COMMENTED** [review]
I like how specific and well thought-out your user stories are. Overall, the user story and user case LGTM except for some minor grammar errors which i think is fine to be overlooked.Good Job!!

**Review by IamZhenHong — APPROVED** [review]
I like how specific and well thought-out your user stories are. Overall, the user story and user case LGTM except for some minor grammar errors which i think is fine to be overlooked.Good Job!!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (wallacexuhanxiao) [reply docs/DeveloperGuide.md:396]: Good suggestion 👍 Added in my latest commit

**Pull request merged**